src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Logging/AbstractRazorLoggerFactory.cs

```diff
@@ -1,41 +1,49 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
-using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Razor.PooledObjects;
 
 namespace Microsoft.CodeAnalysis.Razor.Logging;
 
-internal abstract class AbstractRazorLoggerFactory : IRazorLoggerFactory
+internal abstract partial class AbstractRazorLoggerFactory : IRazorLoggerFactory
 {
-    private readonly ILoggerFactory _loggerFactory;
+    private ImmutableDictionary<string, AggregateLogger> _loggers;
+    private ImmutableArray<IRazorLoggerProvider> _providers;
 
     protected AbstractRazorLoggerFactory(ImmutableArray<IRazorLoggerProvider> providers)
     {
-        _loggerFactory = LoggerFactory.Create(b =>
-        {
-            // We let everything through, and expect individual loggers to control their own levels
-            b.AddFilter(level => true);
-
-            foreach (var provider in providers)
-            {
-                b.AddProvider(provider);
-            }
-        });
+        _providers = providers;
+        _loggers = ImmutableDictionary.Create<string, AggregateLogger>(StringComparer.OrdinalIgnoreCase);
     }
 
     public ILogger CreateLogger(string categoryName)
```

Comment:
Consider renaming this to `GetOrCreateLogger` to indicate that it doesn't necessarily return a newly created logger in each call.